superset/dashboards/filter_sets/api.py

```diff
@@ -0,0 +1,147 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from flask import Response, request, g
+from flask_appbuilder.api import expose, protect, safe, rison, permission_name, merge_response_func, \
+    get_list_schema, API_ORDER_COLUMNS_RIS_KEY, API_LABEL_COLUMNS_RIS_KEY, \
+    API_DESCRIPTION_COLUMNS_RIS_KEY, API_LIST_COLUMNS_RIS_KEY, API_LIST_TITLE_RIS_KEY, ModelRestApi
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+from marshmallow import ValidationError
+from superset import is_feature_enabled
+from superset.commands.exceptions import ObjectNotFoundError
+from superset.dashboards.commands.exceptions import DashboardNotFoundError
+from superset.dashboards.filter_sets.commands.exceptions import FilterSetForbiddenError, FilterSetUpdateFailedError, FilterSetDeleteFailedError, FilterSetCreateFailedError, UserIsNotDashboardOwnerError
+from superset.dashboards.filter_sets.commands.create import CreateFilterSetCommand
+from superset.dashboards.filter_sets.commands.update import UpdateFilterSetCommand
+from superset.dashboards.filter_sets.filters import FilterSetFilter
+from superset.dashboards.filter_sets.schemas import FilterSetPostSchema, FilterSetPutSchema
+from superset.extensions import event_logger
+from superset.models.filter_set import FilterSet
+from superset.views.base_api import BaseSupersetModelRestApi, statsd_metrics
+from superset.dashboards.dao import DashboardDAO
+from superset.dashboards.filter_sets.consts import OWNER_OBJECT_FIELD, DASHBOARD_FIELD, \
+    FILTER_SET_API_PERMISSIONS_NAME, NAME_FIELD, DESCRIPTION_FIELD, OWNER_TYPE_FIELD, OWNER_ID_FIELD, DASHBOARD_ID_FIELD, JSON_METADATA_FIELD
+
+logger = logging.getLogger(__name__)
+
+
+class FilterSetRestApi(BaseSupersetModelRestApi):
+    datamodel = SQLAInterface(FilterSet)
+    resource_name = "dashboard"
+    class_permission_name = FILTER_SET_API_PERMISSIONS_NAME
+    allow_browser_login = True
+    csrf_exempt = True
+    add_exclude_columns = ['id', OWNER_OBJECT_FIELD, DASHBOARD_FIELD]
+    add_model_schema = FilterSetPostSchema()
+    edit_model_schema = FilterSetPutSchema()
+    edit_exclude_columns = ['id', OWNER_OBJECT_FIELD, DASHBOARD_FIELD]
+    list_columns = ['created_on', 'changed_on', 'created_by_fk', 'changed_by_fk', NAME_FIELD,
```

Comment:
we also need to add `method_permission_name = MODEL_API_RW_METHOD_PERMISSION_MAP`